src/events/SDL_mouse.c

```diff
@@ -661,17 +661,47 @@ SDL_SendMouseWheel(SDL_Window * window, SDL_MouseID mouseID, float x, float y, S
     return posted;
 }
 
+static SDL_MouseWheelSource latest_source = SDL_MOUSEWHEEL_SOURCE_LAST;
+
+static void
+SDL_SetPanSource(SDL_MouseWheelSource source) {
+    latest_source = source;
+}
+
+static SDL_MouseWheelSource
+SDL_GetPanLastSource() {
+    return latest_source;
+}
+
+int
+SDL_SendPanDelta(
+        SDL_Window * window,
+        SDL_MouseID mouseID,
+        double delta,
+        SDL_PanAxis axis
+) {
+    SDL_Mouse *mouse = SDL_GetMouse();
+
+    if( SDL_GetEventState(SDL_PANEVENT) == SDL_ENABLE ) {
+        SDL_Event event;
+        event.type = SDL_PANEVENT;
+        event.pan.windowID = mouse->focus ? mouse->focus->id : 0;
+        event.pan.which = mouseID;
+        event.pan.pantype = SDL_PANEVENTTYPE_PAN;
+        event.pan.contents.pan.delta = delta;
+        event.pan.source = SDL_GetPanLastSource();
+        event.pan.axis = axis;
+        return SDL_PushEvent(&event) > 0;
```

Comment:
```suggestion
        return SDL_PushEvent(&event);
```
I think is better so user can handle error with `SDL_GetError`